### PR TITLE
docs(docs-infra): remove hash before link

### LINKED ISSUE #57349: Invalid link
### Describe the problem that you experienced

When i was on the [page](https://angular.dev/api/core/Compiler), and try to click on link ` #compileModuleSync`, on the function `compileModuleAsync: Promise<NgModuleFactory<T>>`, it gives 404

### Enter the URL of the topic with the problem

https://angular.dev/api/core/Compiler

### Describe what you were looking for in the documentation

_No response_

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

_No response_

### Add a screenshot if that helps illustrate the problem

_No response_

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

_No response_

### PR DESCRIPTION
**Pull request by vladboisa**
Remove hash in link for correct view

Fixes #57349

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #57349 

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by JeanMeche — CHANGES_REQUESTED** [review]
This won't fix the issue.  `@link` cannot be used to point to a method. 

`{@link MyClass#myMethod}` should work though.

**Comment by vladboisa** [comment]
> This won't fix the issue. `@link` cannot be used to point to a method.
> 
> `{@link MyClass#myMethod}` should work though.

Done, thank you for a hint!

File: packages/core/src/linker/compiler.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ export class Compiler {
   }
 
   /**
-   * Same as {@link #compileModuleSync} but also creates ComponentFactories for all components.
+   * Same as {@link Compiler#compileModuleSync} but also creates ComponentFactories for all components.
```
Comment (JeanMeche) [inline packages/core/src/linker/compiler.ts:70]: Actually we'll need to go one step further: 
```suggestion
   * Same as {@link Compiler#compileModuleSync compileModuleSync} but also creates ComponentFactories for all components.
```

Comment (vladboisa) [reply packages/core/src/linker/compiler.ts]: Fixed it.
Didn't even think of that, is it to display the link name correctly?
Thank you!

**Review by JeanMeche — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit cacf2b6fc61b9163aaccdda3a6985f72dd55c634.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**